### PR TITLE
Java: Add GeminiTextGenerationService and usage data to chat completion

### PR DESCRIPTION
**Pull request by milderhc**
### Description

Fixes https://github.com/microsoft/semantic-kernel/issues/6580

### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: java/aiservices/google/src/main/java/com/microsoft/semantickernel/aiservices/google/textcompletion/GeminiTextGenerationService.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+                                    textContents.add(
+                                        new TextContent(part.getText(), getModelId(), metadata));
+                                }
+                            });
+                        });
+
+                    return Mono.just(textContents);
+                });
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```
Comment (johnoliver) [inline java/aiservices/google/src/main/java/com/microsoft/semantickernel/aiservices/google/textcompletion/GeminiTextGenerationService.java:89]: make this an SKException

Comment (johnoliver) [reply java/aiservices/google/src/main/java/com/microsoft/semantickernel/aiservices/google/textcompletion/GeminiTextGenerationService.java:89]: should also probably do `return Mono.error(new SKException...`

File: java/semantickernel-api/src/main/java/com/microsoft/semantickernel/orchestration/FunctionResultMetadata.java [hunk 2]
```diff
@@ -50,7 +51,7 @@ public FunctionResultMetadata(CaseInsensitiveMap<ContextVariable<?>> metadata) {
     /**
      * Create a new instance of FunctionResultMetadata.
      */
-    public static FunctionResultMetadata build(String id) {
+    public static FunctionResultMetadata<?> build(@Nullable String id) {
```
Comment (johnoliver) [inline java/semantickernel-api/src/main/java/com/microsoft/semantickernel/orchestration/FunctionResultMetadata.java:54]: Yeah I would leave this non-nullable and generate an id

File: java/aiservices/google/src/main/java/com/microsoft/semantickernel/aiservices/google/textcompletion/GeminiTextGenerationService.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    private GenerativeModel getGenerativeModel(
+        @Nullable PromptExecutionSettings executionSettings) {
+        GenerativeModel.Builder modelBuilder = new GenerativeModel.Builder()
+            .setModelName(getModelId())
+            .setVertexAi(getClient());
+
+        if (executionSettings != null) {
+            if (executionSettings.getResultsPerPrompt() < 1
+                || executionSettings.getResultsPerPrompt() > MAX_RESULTS_PER_PROMPT) {
+                throw new AIException(AIException.ErrorCodes.INVALID_REQUEST,
```
[the remark this answers is not in the document]
Comment (milderhc) [reply java/aiservices/google/src/main/java/com/microsoft/semantickernel/aiservices/google/textcompletion/GeminiTextGenerationService.java:102]: we throw a lot of AIException throughout the codebase, should I update those too?

**Review by johnoliver — APPROVED** [review]

**Pull request merged**